File: Java/Strategy/src/main/java/MiniDuckSimulator.java
package Strategy;

//NOTE -When compiling (javac) use -d ./ to specify root directory for package out
//Running (java) use full package name Strategy.MiniDuckSimulator

public class MiniDuckSimulator {
 
	public static void main(String[] args) {
		MallardDuck	mallard = new MallardDuck();
		RubberDuck rubberDuckie = new RubberDuck();
		DecoyDuck decoy = new DecoyDuck();
 
		ModelDuck model = new ModelDuck();

		mallard.performQuack();
		rubberDuckie.performQuack();
		decoy.performQuack();
   
		model.performFly();	
		model.setFlyBehaviour(new FlyRocketPowered());
		//.performFly();
	}
}

File: Cpp/hfdp/src/strategy/MiniDuckSimulator.cpp
//
// Created by Johnson, Chase on 12/4/16.
//

#include <iostream>
#include "MiniDuckSimulator.h"
#include "MallardDuck.h"
#include "RubberDuck.h"
#include "DecoyDuck.h"
#include "ModelDuck.h"
#include "FlyRocketPowered.h"

void MiniDuckSimulator(void) {
    MallardDuck mallardDuck;
    RubberDuck rubberDuck;
    DecoyDuck decoyDuck;
    ModelDuck modelDuck;

    std::cout << "=========================" << std::endl;
    mallardDuck.display();
    mallardDuck.performQuack();
    mallardDuck.performFly();

    std::cout << "=========================" << std::endl;
    rubberDuck.display();
    rubberDuck.performQuack();
    rubberDuck.performFly();

    std::cout << "=========================" << std::endl;
    decoyDuck.display();
    decoyDuck.performQuack();
    decoyDuck.performFly();

    std::cout << "=========================" << std::endl;
    modelDuck.display();
    modelDuck.performQuack();
    modelDuck.performFly();
    modelDuck.setFlyBehaviour(new FlyRocketPowered());
    modelDuck.performFly();
}
